File: octave/ex0704.m
%> \file 
%> \f$F(x)=x - \sin(x)\f$
%> @author Michel Bierlaire
%> @date Sat Apr  5 19:37:40 2014
%> @ingroup Examples

%> @defgroup Examples 
%> List of examples presented in the book \cite Bier06-book.

%> Example 7.4 in \cite Bier06-book
function [f,g] = ex0704(x)
     f = x - sin(x) ;
     g = 1 - cos(x) ;
endfunction

File: python/examples/ex0704.py
"""
Example 7.4, p. 186

Michel Bierlaire
Fri Dec 10 12:12:49 2021
"""
import numpy as np


def ex0704(x):
    """Example function

    :param x: variable
    :type x: float

    :return: F(x) and F'(x)
    :rtype: float, float
    """
    f = x - np.sin(x)
    g = 1 - np.cos(x)
    return f, g
